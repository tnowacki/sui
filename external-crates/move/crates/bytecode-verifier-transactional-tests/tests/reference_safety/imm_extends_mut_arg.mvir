// tests that function calls require arguments references as disjoint

//# publish

module 0x1.Tester {
    struct T has copy { f: u64 }

    eps1(s: &Self.T, s2: &mut Self.T) {
    label b0:
        abort 0;
    }

    // reject call, i extends s
    t_eps1(s: &mut Self.T) {
        let i: &Self.T;
    label l0:
        i = freeze(copy(s));
        Self.eps1(move(i), move(s));
        return;
    }
}

//# publish

module 0x1.Tester {
    struct T has copy { f: u64 }

    eps2(s2: &mut Self.T, s: &Self.T) {
    label b0:
        abort 0;
    }

    // reject call, i extends s
    t_eps(s: &mut Self.T) {
        let i: &Self.T;
    label l0:
        i = freeze(copy(s));
        Self.eps2(move(s), move(i));
        return;
    }
}
