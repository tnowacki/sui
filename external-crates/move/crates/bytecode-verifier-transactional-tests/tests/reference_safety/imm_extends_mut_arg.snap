---
source: crates/move-transactional-test-runner/src/framework.rs
---
processed 4 tasks

task 0, lines 3-21:
//# publish
Error: Unable to publish module '0000000000000000000000000000000000000000000000000000000000000001::Tester'. Got VMError: {
    major_status: CALL_BORROWED_MUTABLE_REFERENCE_ERROR,
    sub_status: None,
    location: 0x1::Tester,
    indices: [(FunctionDefinition, 1)],
    offsets: [(FunctionDefinitionIndex(1), 5)],
}

task 1, lines 23-41:
//# publish
Error: Unable to publish module '0000000000000000000000000000000000000000000000000000000000000001::Tester'. Got VMError: {
    major_status: CALL_BORROWED_MUTABLE_REFERENCE_ERROR,
    sub_status: None,
    location: 0x1::Tester,
    indices: [(FunctionDefinition, 1)],
    offsets: [(FunctionDefinitionIndex(1), 5)],
}

task 2, lines 43-63:
//# publish
Error: Unable to publish module '0000000000000000000000000000000000000000000000000000000000000001::Tester'. Got VMError: {
    major_status: CALL_BORROWED_MUTABLE_REFERENCE_ERROR,
    sub_status: None,
    location: 0x1::Tester,
    indices: [(FunctionDefinition, 1)],
    offsets: [(FunctionDefinitionIndex(1), 6)],
}

task 3, lines 65-85:
//# publish
Error: Unable to publish module '0000000000000000000000000000000000000000000000000000000000000001::Tester'. Got VMError: {
    major_status: BORROWLOC_EXISTS_BORROW_ERROR,
    sub_status: None,
    location: 0x1::Tester,
    indices: [(FunctionDefinition, 1)],
    offsets: [(FunctionDefinitionIndex(1), 2)],
}
